### DISCUSSION TITLE
Subscription usage

### DISCUSSION THREAD

**Discussion by devnaumov** [DawidWraga/davstack#5] [Q&A] [answered]
Hey! I have a few questions about using the `onChange` method. Does state automatically handle the unsubscribe logic, or do I need to call it manually, for example, from useEffect on component unmount? Also, is it okay to subscribe to certain state changes? For instance, if I have `activeTodo: 1,` I want to run some effect when activeTodo has changed, like this: `todoStore.activeTodo.onChange(someLogic)`. Or there better alternatives?

**Comment by DawidWraga** [comment, marked as answer]
Hey @devnaumov , thanks for your questions. I will improve the docs to make this clearer.

> Is it okay to subscribe to certain state changes? For instance, if I have activeTodo: 1, I want to run some effect when activeTodo has changed, like this: todoStore.activeTodo.onChange(someLogic)

Yes, using `todoStore.activeTodo.onChange(someLogic)` is perfectly fine, assuming that `activeTodo` is state.

Currently, computed values don't have the `onChange` method, but I'll consider adding it in the future.

> Does state automatically handle the unsubscribe logic?

If you are using `createStoreContext`, the `Provider` component automatically unsubscribes from all store effects when the component is unmounted.

It is best practice to always call `onChange` inside `effects`, as it ensures the `onChange` logic is called on the right store instance and can be unsubscribed from correctly.

If you are using a global store, there is no automatic unsubscription. I'm not sure if this is possible or desirable - please let me know if you have any ideas on why or how this could be implemented.

For more details on how `onChange` and `effects` work together, please refer to the updated [onChange and effects documentation](https://www.davstack.com/store/onChange-and-effects).

(I will update the docs now, it should be live shortly)

**Comment by devnaumov** [comment]
Thanks for the clear explanation!
